server/ServerApp.cpp

```diff
@@ -965,6 +966,19 @@ namespace {
                                    << "human player with assigned empire id: " << psd.m_save_game_empire_id;
         }
     }
+
+    /** Replace species selection for any players set to "RANDOM" species, with a species based on @p galaxy_seed */
+    void ValidateSpeciesSelections(std::map<int, PlayerSetupData>& player_setup_data, const std::string& galaxy_seed) {
+        SpeciesManager& sm = GetSpeciesManager();
+        int num_playable_species = sm.NumPlayableSpecies();
+        std::string species_seed = galaxy_seed + "species";
+        for (std::map<int, PlayerSetupData>::value_type& entry : player_setup_data) {
+            if (entry.second.m_starting_species_name == "RANDOM") {
+                entry.second.m_starting_species_name = sm.SequentialPlayableSpeciesName(GetIdxForSeed(num_playable_species,
```

Comment:
what happens if multiple players pick "RANDOM"? do they get the same species?

using this method is OK for galaxy setup settings, but for multiple-evaluations, need to use stateful random functions, perhaps within Python code...